app/controllers/confirmations_controller.rb

```diff
@@ -2,6 +2,10 @@ class ConfirmationsController < Devise::ConfirmationsController
   FLASH_MESSAGE = "Email sent! Please contact support at %<email>s if you are "\
                   "having trouble receiving your confirmation instructions.".freeze
 
+  def show
+    super { sign_in(resource) if resource.errors.empty? && resource.creator? }
```

Comment:
Not everyone may agree with this but I'd probably move this method's code from `Devise::ConfirmationsController` over and make the necessary changes. This is quite an important action with potential security implications, I'd rather see the whole flow than in one place, it's too easy to get confused or overlook something when this logic is spread over two files.